torch/fx/graph.py

```diff
@@ -561,7 +561,11 @@ def get_attr(self, qualified_name: str, type_expr: Optional[Any] = None) -> Node
         def _get_attr_reference_exists(mod: torch.nn.Module, qualified_name: str) -> bool:
```

Comment:
This whole function looks like it was written with the assumption that `Module.get_submodule` returns None if that submodule does not exist, but that is not the case if you look at the function:

https://github.com/pytorch/pytorch/blob/992dad18552883224016d53429c48f2e932651cf/torch/nn/modules/module.py#L390

Alternative fixes would be:
1) Make get_submodule conform to that behavior
2) Add a `Module.find_submodule` method that `get_submodule` would be implemented in terms of. Then, `get_submodule` is just a wrapper that additionally throws if it gets a None value from `find_submodule`

I'm not sure if we want to do more intrusive changes to `nn.Module` so I'm going to just approve this PR for now

Reply:
But given the above, can you update `submod` to be typed ad `nn.Module` and delete the code below that handles if `submod` is None?